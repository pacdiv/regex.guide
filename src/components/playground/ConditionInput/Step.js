import React from "react"
import PropTypes from "prop-types"

import { StepContainer } from "./Step.style"

function ConditionInputStep({ children, currentStep, title }) {
  return (
    <StepContainer>
      <p>
        <i>Step {currentStep}</i>
        <br />
        {title}
      </p>
      {children}
    </StepContainer>
  )
}

ConditionInputStep.propTypes = {
  children: PropTypes.node,
  currentStep: PropTypes.number,
  title: PropTypes.string,
}

export default ConditionInputStep
