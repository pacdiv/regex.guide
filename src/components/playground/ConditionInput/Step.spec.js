import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import Step from "./Step"

test("Step component", async () => {
  const { container } = render(
    <Step currentStep={1} title="Best way to debug code?">
      <div>
        <p>Delete it!</p>
      </div>
    </Step>
  )

  expect(container).toMatchSnapshot()
})
