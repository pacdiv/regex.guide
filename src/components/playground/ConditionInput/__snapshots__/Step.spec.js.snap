// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step component 1`] = `
<div>
  <div
    class="css-1bvw1ww"
  >
    <p>
      <i>
        Step 1
      </i>
      <br />
      Best way to debug code?
    </p>
    <div>
      <p>
        Delete it!
      </p>
    </div>
  </div>
</div>
`;
